03-rnaseq/pathway-analysis_rnaseq_03_gsva.Rmd

```diff
@@ -0,0 +1,475 @@
+In this chunk of code, we will not provide a specific model to the `design` argument because we are not performing a differential expression analysis.
+
+```{r}
+# Create a `DESeqDataSet` object
+dds <- DESeqDataSetFromMatrix(
+  countData = expression_df, # Our prepped data frame with counts
+  colData = metadata, # Data frame with annotation for our samples
+  design = ~1 # Here we are not specifying a model
+)
+```
+
+## Perform DESeq2 normalization and transformation
+
+We often suggest normalizing and transforming your data for various applications including for GSVA. 
+We are going to use the `vst()` function from the `DESeq2` package to normalize and transform the data.
+For more information about these transformation methods, [see here](https://alexslemonade.github.io/refinebio-examples/03-rnaseq/00-intro-to-rnaseq.html#deseq2-transformation-methods). 
+
+```{r}
+# Normalize and transform the data in the `DESeqDataSet` object using the `vst()`
+# function from the `DESEq2` R package
+dds_norm <- vst(dds)
+```
+
+At this point, if your data set has any outlier samples, you should look into removing them as they can affect your results. 
+For this example data set, we will skip this step (there are no obvious outliers) and proceed.
+
+But now we are ready to format our dataset for input into `gsva::gsva()`.
+We need to extract the normalized counts to a matrix and make it into a data frame so we can use with tidyverse functions later. 
+
+```{r}
+# Retrieve the normalized data from the `DESeqDataSet`
+normalized_counts <- assay(dds_norm) %>%
+  as.data.frame() %>% # Make into a data frame
+  tibble::rownames_to_column("ensembl_id") # Make Gene IDs into their own column
+```
+
+**Review**
+
+### Import Gene Sets
+
+The function that we will use to run GSVA wants the gene sets to be in a list where each pathway has its own gene list inside of it. 
+
+Here we are obtaining the pathway information from the main function of the [`msigdbr`](https://cran.r-project.org/web/packages/msigdbr/index.html) package [@Igor2020].
+Because we are using human data in this example, we supply the formal organism name to the `species` argument. 
+We will want only the hallmark pathways, so we use the `category = "H"` argument.
+
+```{r}
+hallmark_gene_sets <- msigdbr::msigdbr(
+  species = "Homo sapiens", # Can change this to what species you need
+  category = "H" # Only hallmark genes
+)
+```
+
+`hallmark_gene_sets` has which genes are a part of which pathway in a data frame format.
+For this example we will use Entrez IDs (but note that there's gene symbols we could just as easily use).
+The info we need is in two columns: `entrez_gene` contains the gene ID and `gs_name` contains the name of the pathway that the gene is a part of. 
+
+To make this into the list format we need, we can use `split()` function.
+Here we want `entrez_gene` to be made into smaller lists for each pathway.
```

Comment:
I don't think this is quite correct (this comment extends to the in line comments in the code chunk below, too). We want a list where each element is a vector that contains the Entrez gene IDs that are in a particular gene set.

I would also probably use wording like "we want the Entrez IDs, stored in the `entrez_gene` column, to be..."

Reply:
See how you feel about the re-wording.